UefiDbgExt/swdebug.cpp

```diff
@@ -152,3 +152,235 @@ reboot (
   EXIT_API ();
   return S_OK;
 }
+
+ULONG64 ReadTaggedValue (
+  PCSTR   Response,
+  PCSTR   Tag
+  )
+{
+  PCSTR TagPos = strstr(Response, Tag);
+  if (TagPos != NULL) {
+    PCSTR ValuePos = TagPos + strlen(Tag);
+    return strtoull(ValuePos, NULL, 16);
+  }
+
+  return 0;
+}
+
+HRESULT CALLBACK
+reload (
+  PDEBUG_CLIENT4  Client,
+  PCSTR           args
+  )
+{
+  HANDLE hFile = INVALID_HANDLE_VALUE;
+  CHAR   Command[512];
+  PSTR   Response;
+  BYTE*  ImageBuffer = NULL;
+  BYTE*  CompressedImage = NULL;
+  BOOLEAN NoGo = FALSE;
+  INIT_API ();
+
+  if (RUST != gUefiEnv) {

```

Comment:
Not for this PR, but in the future we may want to make this PATINA_DXE or something.